corehq/apps/translations/app_translations/upload_module.py

```diff
@@ -97,8 +97,50 @@ def update(self, rows):
         if self.search_again_label:
             self._update_translation(self.search_again_label, self.module.search_config.again_label)
 
+        self._update_case_search_labels(rows)
+
         return self.msgs
 
+    def _update_case_search_labels(self, rows):
+        properties = self.module.search_config.properties
+        displays = [row for row in self.condensed_rows if row['list_or_detail'] == 'case_search_display']
+        hints = [row for row in self.condensed_rows if row['list_or_detail'] == 'case_search_hint']
+        if len(displays) != len(hints) or len(displays) != len(properties):
+
+            message = _(
+                'Expected {expected_count} case_search_display and case_search_hint '
+                'properties in  menu {index}, found {actual_label_count} for case_search_display and '
+                '{actual_hint_count} for case_search_hint'
+                'No Case Search config properties for menu {index} were updated.'
+            ).format(
+                expected_count=len(properties),
+                actual_label_count=len(displays),
+                actual_hint_count=len(hints),
+                index=self.module.id + 1,
+            )
+            self.msgs.append((messages.error, message))
+        else:
+            for display_row, prop in itertools.chain(zip(displays, properties)):
+                if display_row.get('case_property') != prop.name:
+                    message = _('A row for menu {index} has an unexpected case search property "{field}". '
+                                'Case properties must appear in the same order as they do in the bulk '
+                                'app translation download. No translations updated for this row.').format(
+                                    index=self.module.id + 1,
+                                    field=display_row.get('case_property', ""))
+                    self.msgs.append((messages.error, message))
+                    continue
+                self._update_translation(display_row, prop.label)
+            for hint_row, prop in itertools.chain(zip(hints, properties)):
+                if hint_row.get('case_property') != prop.name:
+                    message = _('A row for menu {index} has an unexpected case search property "{field}". '
```

Comment:
Should this message specify that it's a hint row that's the issue?

Reply:
Sure!